orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/utils/TrafficGuard.java

```diff
@@ -231,7 +260,8 @@ public void verifyTrafficRemoval(
     TargetServerGroup someServerGroup = serverGroupsGoingAway.stream().findAny().get();
     Location location = someServerGroup.getLocation();
 
-    if (!hasDisableLock(someServerGroup.getMoniker(), account, location)) {
+    Front50Cache front50Cache = new Front50Cache(this);
```

Comment:
doesn't creating a new cache here defeat some of the cache benefit vs passing it in from line 248?

Reply:
the intent of the cache is it should just live through one public call path through TrafficGuard, so this one is at the start of one of the public verifyTrafficRemoval signatures that eventually calls the same verifyTrafficRemovalInternal as line 248 is calling

There was one case where (for instance termination) the entrypoint from one public method was calling into another public method so that is where I introduced a private method that allows propagating the cache along